### PR TITLE
Rotate translate regions and fixes

### PR DESCRIPTION
**Pull request by jgsdavies**
Arc currently need start,end,centre,radius. On first inspection it seems that only centre or radius is required to fully define the arc however this is not the case:

2 valid solutions for start/end/centre
image

These are both valid solutions for arc with start/end/rad
image

We have some options:
1. force user to specify direction of arc
2. Assume arc smaller than 180 degrees

I've tried to implement idea 2 whilst still allowing user to force arc with greater than 180 deg by using all 4 params.

Nice side effect of these changes is that centre coord can become a property rather than a class object. This means rotation/translation can be handled by parent Entity class in same way for both arc and Line.

Hoping this makes the most sense as rounding corners etc is very simple when you only have to provide start/end/radius

### CODE REVIEW

File: src/ansys/motorcad/core/geometry.py [hunk 1]
```diff
@@ -554,7 +552,11 @@ def rotate(self, centre_point, angle):
         centre_point : Coordinate
             point to rotate Coordinate around.
```
Comment (PipKat) [inline src/ansys/motorcad/core/geometry.py:553]:
```suggestion
            Point to rotate coordinate around.
```

File: src/ansys/motorcad/core/geometry.py [hunk 2]
```diff
@@ -435,13 +409,37 @@ def edit_point(self, old_coordinates, new_coordinates):
...
             if entity.end == old_coordinates:
                 entity.end = deepcopy(new_coordinates)
-            if isinstance(entity, Arc):
-                # Check Arc is still valid
-                _ = entity.centre
+                edited = True
+
+            if edited and isinstance(entity, Arc):
+                # Draw line between arc start/end
+                # get centre point of that line
```
Comment (PipKat) [inline src/ansys/motorcad/core/geometry.py:423]:
```suggestion
                # Get center point of that line
```

File: src/ansys/motorcad/core/geometry.py [hunk 3]
```diff
@@ -897,14 +876,29 @@ def get_line_intersection(self, line):
             return None
 
 
-class _BaseArc(Entity):
-    """Internal class to allow creation of Arcs."""
+class Arc(Entity):
+    """Python representation of Motor-CAD arc entity based upon start, end, centre and radius.
```
Comment (PipKat) [inline src/ansys/motorcad/core/geometry.py:880]:
```suggestion
    """Python representation of Motor-CAD arc entity based upon start, end, centre, and radius.
```

**Review by JackB-Ansys — APPROVED** [review]

File: src/ansys/motorcad/core/geometry.py (hunk 2 above)
Comment (jgsdavies) [reply src/ansys/motorcad/core/geometry.py]: Motor-CAD API methods are British English so trying to have some consistency. Most methods are pre PyAnsys so decided it's too big a change for customers to switch now

File: src/ansys/motorcad/core/geometry.py (hunk 1 above)
Comment (jgsdavies) [reply src/ansys/motorcad/core/geometry.py:555]: Coordinate is a class name so keeping this capitalised

Comment (jgsdavies) [reply src/ansys/motorcad/core/geometry.py:555]:
```suggestion
            Point to rotate Coordinate around.
```

File: src/ansys/motorcad/core/geometry.py [hunk 4]
```diff
@@ -552,7 +552,7 @@ def rotate(self, centre_point, angle):
         Parameters
         ----------
         centre_point : Coordinate
-            point to rotate Coordinate around.
+            Point to rotate Coordinate around.
```
Comment (PipKat) [inline src/ansys/motorcad/core/geometry.py:555]:
```suggestion
            Point to rotate coordinate around.
```
Only capitalize proper nouns or the first letter of the initial word (as you did).

**Review by JackB-Ansys — APPROVED** [review]

**Review by JackB-Ansys — APPROVED** [review]

**Pull request merged**